packages/core/src/shared/clients/docdbClient.ts

```diff
@@ -3,14 +3,41 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import * as vscode from 'vscode'
-import _ from 'lodash'
-import { AWSError, DocDB, DocDBElastic } from 'aws-sdk'
+import globals from '../extensionGlobals'
 import { getLogger } from '../logger'
+import { getUserAgent } from '../telemetry/util'
+import { ToolkitError } from '../errors'
 import { InterfaceNoSymbol } from '../utilities/tsUtils'
+import { DocDBClient, DocDBClientConfig, DBCluster, DescribeDBClustersCommand } from '@aws-sdk/client-docdb'
 
 export type DocumentDBClient = InterfaceNoSymbol<DefaultDocumentDBClient>
 
 export class DefaultDocumentDBClient {
     public constructor(public readonly regionCode: string) {}
+
+    public async getClient(): Promise<DocDBClient> {
+        const credentials = await globals.awsContext.getCredentials()
+        const config: DocDBClientConfig = {
+            customUserAgent: await getUserAgent({ includePlatform: true, includeClientId: true }),
+            credentials: credentials,
+            region: this.regionCode,
+        }
+        return new DocDBClient(config)
+    }
+
+    public async listClusters(): Promise<DBCluster[]> {
+        getLogger().debug('ListClusters called')
+        const client = await this.getClient()
+
+        try {
+            const input = {
+                Filters: [{ Name: 'engine', Values: ['docdb'] }],
+            }
+            const command = new DescribeDBClustersCommand(input)
+            const response = await client.send(command)
+            return response.DBClusters ?? []
+        } catch (e) {
+            throw ToolkitError.chain(e, 'Failed to get DocumentDB clusters')
+        }
+    }
```

Comment:
This logic seems like it should be testable and would benefit from tests. Is it possible to restructure to allow testing of this logic?

Reply:
This is just a lightweight wrapper around the v3 sdk client, not sure how much benefit tests would add here.